File: exercism/csharp/log-levels/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
        return logLine
            .Replace($"[{LogLevel(logLine).ToUpper()}]: ", "")
            .Replace("\r", "")
            .Replace("\n", "")
            .Replace("\t", "")
            .Trim();
    }

    public static string LogLevel(string logLine)
    {
        return logLine.Substring(logLine.IndexOf("[") + 1, logLine.IndexOf("]") - 1).ToLower();
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
    }
}

File: exercism/java/log-levels/src/main/java/LogLevels.java
public class LogLevels {

    public static String message(String logLine) {
        return logLine
                .replace(new StringBuilder("[").append(logLevel(logLine).toUpperCase()).append("]: ").toString(), "")
                .replace("\r", "")
                .replace("\n", "")
                .replace("\t", "")
                .trim();
    }

    public static String logLevel(String logLine) {
        return logLine.substring(logLine.indexOf("[") + 1, logLine.indexOf("]")).toLowerCase();
    }

    public static String reformat(String logLine) {
        return new StringBuilder(message(logLine)).append(" (").append(logLevel(logLine)).append(")").toString();
    }
}

